File: py-lunamodel/src/lm/expression.py
from lm._core import PyExpression
from .environment import Environment


class Expression:
    """ """

    _expr: PyExpression


    def __init__(self, env: Environment | None = None) -> None:
        if env is None:
            self._expr = PyExpression()
        else:
            self._expr = PyExpression(env._env)

File: pyo3-lunamodel/src/expression.rs
use lunamodel_python::prelude::PyExprContent;

#[repr(transparent)]
pub struct PyExpression(pub PyExprContent);

capsule_wrapper! {
    wrapper: PyExpression,
    public: Expression,
    inner: PyExprContent,
    attr: "_expr",
    from_py: "_from_pyexpr",
}
